### ISSUE TITLE
17 Test fails with 0.35.0 probably related to test data fetching.

### ISSUE THREAD

**Issue by TheChymera** [dandi/dandi-cli#878]
This is the local test summary, I assume it could be related to data-fetching not working in the sandbox, in which case it would be cool if there was a direct test data archive which can be fetched explicitly (many of the tests have `urllib3.exceptions.NewConnectionError` in the full traceback) — or it could be something in addition to that as well, any info is welcome :)

```
========================================================= short test summary info ==========================================================
FAILED dandi/cli/tests/test_command.py::test_smoke[command1] - AssertionError: Exited abnormally. out=Error: Ignore list for validate() c...
FAILED dandi/cli/tests/test_ls.py::test_smoke[json] - KeyError: 'session_id'
FAILED dandi/cli/tests/test_ls.py::test_smoke[json_pp] - KeyError: 'session_id'
FAILED dandi/cli/tests/test_ls.py::test_smoke[json_lines] - KeyError: 'session_id'
FAILED dandi/cli/tests/test_ls.py::test_smoke[yaml] - KeyError: 'session_id'
FAILED dandi/cli/tests/test_ls.py::test_smoke_local_schema - KeyError: 'digest'
FAILED dandi/tests/test_metadata.py::test_get_metadata - TypeError: Ignore list for get_metadata() contains an unexpected keyword argumen...
FAILED dandi/tests/test_metadata.py::test_species - requests.exceptions.ConnectionError: HTTPConnectionPool(host='purl.obolibrary.org', p...
FAILED dandi/tests/test_organize.py::test_ambiguous - assert 1 == 0
FAILED dandi/tests/test_pynwb_utils.py::test_nwb_has_external_links - TypeError: Ignore list for nwb_has_external_links() contains an une...
FAILED dandi/tests/test_utils.py::test_get_instance_dandi_with_api - ValueError: Invalid version string: '0+unknown'
FAILED dandi/tests/test_utils.py::test_get_instance_url - ValueError: Invalid version string: '0+unknown'
FAILED dandi/tests/test_utils.py::test_get_instance_cli_version_too_old - ValueError: Invalid version string: '0+unknown'
FAILED dandi/tests/test_utils.py::test_get_instance_bad_cli_version - ValueError: https://example.dandi/ returned an incorrectly formatte...
FAILED dandi/tests/test_validate.py::test_validate_simple1 - TypeError: Ignore list for validate() contains an unexpected keyword argumen...
FAILED dandi/tests/test_validate.py::test_validate_simple2 - TypeError: Ignore list for validate() contains an unexpected keyword argumen...
FAILED dandi/tests/test_validate.py::test_validate_simple2_new - TypeError: Ignore list for validate() contains an unexpected keyword arg...
ERROR dandi/tests/test_fixtures.py::test_organized_nwb_dir - AssertionError: [Parallel(n_jobs=-1)]: Using backend LokyBackend with 12 con...
ERROR dandi/tests/test_fixtures.py::test_organized_nwb_dir2 - AssertionError: [Parallel(n_jobs=-1)]: Using backend LokyBackend with 12 co...
========================================== 17 failed, 189 passed, 140 skipped, 2 errors in 18.16s ==========================================
```

Here is the full build and test log: https://ppb.chymera.eu/6ff00d.log

**Comment by jwodder** [comment]
@TheChymera

* Many of the failures seem to be caused by "TypeError: Ignore list for \* contains an unexpected keyword argument '_cache_fingerprint'", which indicates that you're using an old version of fscacher that doesn't work with newer versions of joblib.  Upgrade fscacher and try again.
* The tests that failed with "ValueError: Invalid version string: '0+unknown'" indicate that you're using a copy of dandi-cli without the relevant Git history needed to determine a version number.  You need to install dandi-cli from a full clone, an sdist, or a wheel, not a shallow clone.
* I see only one test with `urllib3.exceptions.NewConnectionError` in the traceback, and that was because it wasn't marked with `skipif_no_network`; I've created a PR to fix it: #886.
* I'm not certain about the remaining failures; see what happens to them after the above are taken care of.

**Comment by yarikoptic** [comment]
I think @jwodder identified all possible causes for failing tests, and such that there is nothing to be done on dandi-cli side.  If there is anything to fix, @TheChymera please reopen or file new dedicated issue(s).

**Comment by TheChymera** [comment]
Can confirm, issues solved as of the PyPI `0.45.1` release.
